stackit/internal/services/cdn/customdomain/resource.go

```diff
@@ -160,7 +208,7 @@ func (r *customDomainResource) Create(ctx context.Context, req resource.CreateRe
 		return
 	}
 
-	err = mapCustomDomainFields(waitResp, &model)
+	err = mapCustomDomainFields(waitResp, &model, respPutCustomDomain.Certificate)
```

Comment:
Based on the suggestion of the comment before, you would need to make here an GET call and pass the response to the map function.

```suggestion
	respCustomDomain, err := r.client.GetCustomDomainExecute(ctx, projectId, distributionId, name)
	if err != nil {
		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating CDN custom domain", fmt.Sprintf("Calling API: %v", err))
		return
	}
	err = mapCustomDomainFields(respCustomDomain, &model)
```